### ISSUE TITLE
Emit diagnostic for explicit subclass of protocol that doesn't implement declared (but unassigned) variable

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2878]
This is a tracking issue. For the original, see https://github.com/microsoft/pylance-release/issues/2254.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.211, which I just published. It will also be included in the next release of pylance.

**Comment by thomkeh** [comment]
Should the same be done for empty functions defined in a Protocol? If you have code like this:

```python
from typing import Protocol

class Sized(Protocol):
    def __len__(self) -> int: ...

class A(Sized):
    def __lne__(self) -> int:  # (misspelled "__len__")
        return 0
```
it's of course technically correct because `A` inherits the default implementation (which is just the empty function here) from `Sized` but this is clearly not what the programmer intended. There is actually also a type error here because `len(A)` will return `None` instead of the declared `int`.

FWIW, mypy also doesn't report an error with the above.

**Comment by erictraut** [comment]
Yeah, that's a reasonable feature request. There are some necessarily limitations though. If the base class protocol is defined in a stub file, pyright can't tell whether its implementation was omitted because it's not provided or because implementations are always omitted in stub files. But if the protocol is defined in a ".py" file, pyright can enforce this.

This will be included in the next release.
